test/test_torch.py

```diff
@@ -3094,59 +3094,54 @@ def _test_broadcast(self, cast):
                 # map and map2 are not implementd on CUDA tensors
                 continue
 
-            # TODO: fix masked_scatter and masked_fill broadcasting
-            if hasattr(large_expanded, fn) and fn not in ['masked_scatter', 'masked_fill']:
+            def specialized(myfn, t1, t2, t3):
+                if fn == "lerp":
+                    return myfn(t1, t2, 0.5)
+                elif fn == "masked_select":
+                    return myfn(t1, t2 < 0)
+                elif fn == "masked_scatter":
+                    expanded = torch.broadcast_tensors(t1, t2)[0]
+                    return myfn(t1, t2 < 0.5, torch.ones_like(expanded))
+                elif fn == "masked_fill":
+                    return myfn(t1, t2 < 0.5, 1.0)
+                elif fn in fns_3_args:
+                    return myfn(t1, 1.0, t2, t3)
+                else:
+                    return myfn(t1, t2)
+
+            if hasattr(large_expanded, fn):
                 # run through tensor versions of functions
                 # and verify fully expanded inputs give same results
                 expanded = {large: large_expanded, small: small_expanded, small2: small2_expanded}
 
-                def tensorfn(myfn, t1, t2):
-                    if fn == "lerp":
-                        return myfn(t1, 0.5)
-                    elif fn == "masked_select":
-                        return myfn(t1 < 0)
-                    elif fn in fns_3_args:
-                        return myfn(1, t1, t2)
-                    else:
-                        return myfn(t1)
+                def prepend_unused_arg(f):
+                    def wrapper(_, *args):
+                        return f(*args)
+                    return wrapper
 
                 # test various orders
                 for first, second, third in [(large, small, small2), (small, large, small2),
                                              (small2, small, large), (small2, large, small)]:
                     if first is None:
                         break  # ignore last iter when small2 is None
-                    method_expanded = getattr(expanded[first], fn)
-                    method = getattr(first, fn)
-                    r1 = tensorfn(method_expanded, expanded[second], expanded[third])
-                    r2 = tensorfn(method, second, third)
+                    method_expanded = prepend_unused_arg(getattr(expanded[first], fn))
+                    method = prepend_unused_arg(getattr(first, fn))
+                    r1 = specialized(method_expanded, expanded[first], expanded[second], expanded[third])
+                    r2 = specialized(method, first, second, third)
                     self.assertEqual(r1, r2)
 
             # now for torch. versions of functions
             if hasattr(torch, fn):
                 fntorch = getattr(torch, fn)
                 expanded = {large: large_expanded, small: small_expanded, small2: small2_expanded}
 
-                def torchfn(t1, t2, t3):
```

Comment:
This is merged with the `def tensorfn` below and it is now the `def specialized`